### DISCUSSION TITLE
Sunsetting Mu in 2025

### DISCUSSION THREAD

**Discussion by ntoll** [mu-editor/mu#2535] [General]
See the announcement here: https://madewith.mu/mu/users/2024/12/10/retirement-plans.html

Feedback, ideas and constructive criticism welcome in this discussion.

**Comment by jepler** [comment]
By no means speaking officially for Adafruit/CircuitPython, but I just wanted to thank you for providing Mu in the first place & providing a thoughtful timeline for its sunset now.

Mu has been our suggested editor for CircuitPython for as long as I've been involved in the project; choosing what to recommend now will be interesting for us.

Be well & do good stuff in the future!

**Reply by Timeline8** [reply]
Second that. I have only started on my CircuitPython journey less than a year ago (Feb/Mar 2024 I think) and have been using MU the whole time. As a non-programmer wanting to learn how to program microcontroller boards to do stuff for me, I was pleased to find MU so stupid-easy to use that I wasn't spending time on learning to use an editor first. I see a lot of posts from people using things like Thonny and half the time the issue turns out to be the editor not configured properly, wrong libraries loaded, or pointing to running on the wrong device (trying to run on the computer vs. the MCU board), MU made all that irrelevant. So thanks from me for something that helped me get going quickly learning the programming language and not the editor environment.

**Reply by jepler** [reply]
On the Adafruit Discord we've opened a thread discussing "successor to mu" in the context of CircuitPython: https://discord.com/channels/327254708534116352/537365702651150357/1316045877076426764

**Comment by Timeline8** [comment]
I do have ONE suggestion I would like to see in the final releases of MU if it is easy to do. The one thing that has always irked me about MU is when my development board ejects (especially due to a crash and the eject was not my doing), the serial window immediately closes and the history is lost. When trying to solve a crash & eject problem is harder when you can not see the potentially telling final serial message or debug print statement from my code. If the board disconnects I would like the window to simply "go dead" but remain open and displaying the history.
